src/cbc_sdk/platform/base.py

```diff
@@ -957,6 +959,11 @@ def _search(self, start=0, rows=0):
             self._processed_segments = result.get("processed_segments", 0)
             self._count_valid = True
             if self._processed_segments != self._total_segments:
+                retry_counter = 0 if self._processed_segments > last_processed_segments else retry_counter + 1
+                last_processed_segments = max(last_processed_segments, self._processed_segments)
+                if retry_counter == MAX_EVENT_SEARCH_RETRIES:
+                    raise TimeoutError(url, resp.status_code, "excessive number of retries in event query")
+                time.sleep(1 + retry_counter / 10)
```

Comment:
Any particular reason you chose a divisor of 10? 100ms doesn't seem like an appreciable difference in sleep times to me

Reply:
Alex mentioned that the overall wait should be 10-15 seconds.  This was an easy way to add a slight bit of backoff to the individual wait cycles; total wait would then be 14.5 seconds.